digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/PolygonTools.scala

```diff
@@ -0,0 +1,91 @@
+package fi.liikennevirasto.digiroad2.util
+
+import java.util.Properties
+
+import com.vividsolutions.jts.geom._
+import fi.liikennevirasto.digiroad2.asset.BoundingRectangle
+import fi.liikennevirasto.digiroad2.user.UserProvider
+import org.geotools.geometry.jts.GeometryBuilder
+import fi.liikennevirasto.digiroad2.Point
+import com.vividsolutions.jts.io.WKTReader
+import scala.collection.mutable.ListBuffer
+
+/**
+  * Tools related to polygons
+  */
+class PolygonTools {
+  val geomFact = new GeometryFactory()
+  val geomBuilder = new GeometryBuilder(geomFact)
+  lazy val properties: Properties = {
+    val props = new Properties()
+    props.load(getClass.getResourceAsStream("/digiroad2.properties"))
+    props
+  }
+  lazy val userProvider: UserProvider = {
+    Class.forName(properties.getProperty("digiroad2.userProvider")).newInstance().asInstanceOf[UserProvider]
+  }
+
+  /**
+    *
+    * @param geometry jts Geometry
+    * @param boundingBox  BoundingRectangle
+    * @return returns Sequence of JTS Polygons that are with in bounding box
+    */
+  def geometryInterceptorToBoundingBox(geometry: Geometry, boundingBox: BoundingRectangle): Seq[Polygon] = {
+    var listGeom = ListBuffer.empty[Polygon]
+    val leftBottomP = boundingBox.leftBottom
+    val rightTopP = boundingBox.rightTop
+    val leftTopP = Point(leftBottomP.x, rightTopP.y)
+    val rightBottom = Point(rightTopP.x, leftBottomP.y)
+    val BoundingBoxAsPoly = geomBuilder.polygon(leftTopP.x, leftTopP.y, rightTopP.x, rightTopP.y, rightBottom.x, rightBottom.y, leftBottomP.x, leftBottomP.y)
+    val intersectionGeometry = geometry.intersection(BoundingBoxAsPoly)
+    if (intersectionGeometry.getGeometryType.toLowerCase.startsWith("polygon")) {
+      listGeom += intersectionGeometry.asInstanceOf[Polygon]
```

Comment:
yea, at first i planned on making multipolygon-> polygon sequence conversion in that method but now that it is in different method we don't need buffered list there anymore